python/HeaderService/camera_coords.py

```diff
@@ -253,62 +287,170 @@ def mosaic_ITL(self, Sx, Sy):
+                   'CRPIX2A': 0,
+                   'CRVAL1A': Sx*(self.dimv + 1),
+                   'CRVAL2A': self.dimh + 1 - self.preh}
+        return wcs
+
+    def ccd2segment(self, xc, yc, Sx, Sy):
+        """
+        This function maps CCD (a.k.a. sensor) coordinates (xc,yc) into
+        Segment coordinates (xs,ys) using the information of LCA-13501 and the
+        (Sx,Sy) components of the segment location designator
+
+        The S segment location designator (00,01...07,10,11...17) in a CCD can
+        be decomposed into their x and y index values as described in LCA-13501
+        (see section 9.2), such as S = 10*Sx + Sy.
+
+        This transformation comes solving the WCS matrix that maps CCDs to
+        Segments in section 9.6
+
+        Parameters
+        ----------
+        xc : float
+             The x value on CCD coordinates.
+        yc : float
+             The y value on CCD coordinates.
+        Sx: int
+            The x component of the segment location Designator
+        Sy: int
+            The y component of the segment location Designator
+
+        Returns
+        -------
+        xs: float
+            The x value on on Segment coordinates.
+        ys: float
+            The y value on on Segment coordinates.
+        """
+
+        if self.vendor == 'E2V':
+            xs = (yc - Sx*(self.dimh + 1) - Sy*self.dimh)/(1. - 2.*Sx) + self.preh
+        elif self.vendor == 'ITL':
+            xs = 1 - yc + self.dimh*(1 + Sy) - self.preh
+        ys = (xc - Sx*(2*self.dimv + 1))/(1. - 2.*Sx)
+        return xs, ys
+
+    def segment2ccd(self, xs, ys, Sx, Sy):
+        """
+        This function maps Segment coordinate (xs,ys) into CCD coordinates
+        (xs,ys) using the information of LCA-13501 and the (Sx,Sy) components
+        of the segment location designator
+
+        The S segment location designator (00,01...07,10,11...17) in a CCD can
+        be decomposed into their x and y index values as described in LCA-13501
+        (see section 9.2), such as S = 10*Sx + Sy.
+
+        This transformation comes solving the WCS matrix that maps CCDs to
+        Segments in section 9.6
+
+        Parameters
+        ----------
+        xs : float
+             The x value on Segment coordinates.
+        ys : float
+             The y value on Segment coordinates.
+        Sx: int
+            The x component of the segment location Designator
+        Sy: int
+            The y component of the segment location Designator
+
+        Returns
+        -------
+        xc: float
+            The x value on CCD coordinates.
+        yc: float
+            The y value on CCD coordinates.
+        """
+
+        if self.vendor == 'E2V':
+            yc = xs*(1-2*Sx)+Sx*(self.dimh+1)+Sy*self.dimh+(2*Sx-1)*self.preh
+        elif self.vendor == 'ITL':
+            yc = 1 - xs + self.dimh*(1 + Sy) - self.preh
+        xc = (1 - 2*Sx)*ys + Sx*(2*self.dimv + 1)
+        return xc, yc
 
```

Comment:
I thought we'd decided to remove per-segment WCS entries, as agreed with Tony.  What's going on here?

Reply:
Same as above, we are not using this function. I kept in the code in case we want to use it in the future. If you want it removed, just say remove it.